File: Least Prime Factor[07-June-2023]/solution.py
#  >> PYTHON CODE <<

class Solution:
    def leastPrimeFactor (self, n):
        primes=[-1]*(n+1)
        
        primes[0]=0
        primes[1]=1
        
        i=2
        while(i<=n):
            j=i
            while(j<=n):
                if(primes[j]==-1):
                    primes[j]=i
                
                j+=i
            
            i+=1

        return primes

File: Least Prime Factor[07-June-2023]/solution.cpp
//  >> C++ CODE <<

vector<int> leastPrimeFactor(int n) {
        vector<int> primes(n + 1, -1);
        primes[0] = 0;
        primes[1] = 1;
    
        int i = 2;
        while (i <= n) {
            int j = i;
            while (j <= n) {
                if (primes[j] == -1) {
                    primes[j] = i;
                }
                j += i;
            }
            i++;
        }
    
        return primes;

    }